test/test_grok.py

```diff
@@ -313,7 +313,7 @@ def test_v1_tokenize_text_with_valid_request(self):
         expected_response = schemas.TokenizeTextResponse(**mock_response_data)
         v1_tokenize_text_response = test_utils.MockResponse(mock_response_data)
         self.mock_requests.post = mock.Mock(
-            spec_set=[], return_value=v1_tokenize_text_response
+            spec_set=[], side_effect=(v1_tokenize_text_response,), return_value=None
```

Comment:
The mock setup for `side_effect` is added but `return_value=None` is also set, which will override the side effect. The mock will always return `None` instead of the intended mock response.